test/system/flags/image-name-and-tag.spec.ts

```diff
@@ -0,0 +1,60 @@
+import { DepGraph } from "@snyk/dep-graph";
+import { join as pathJoin } from "path";
+import { scan } from "../../../lib/index";
+
+function getFixture(fixturePath: string): string {
+  return pathJoin(__dirname, "../../fixtures", fixturePath);
+}
+
+describe("imageNameAndTag tests", () => {
+  it("it overrides name and version when reading a docker archive", async () => {
+    const fixturePath = getFixture("docker-archives/docker-save/nginx.tar");
+    const path = `docker-archive:${fixturePath}`;
+    const imageNameAndTag = "nginx:1.23.4";
+
+    const pluginResult = await scan({
+      path,
+      imageNameAndTag,
+    });
+
+    const depGraph: DepGraph = pluginResult.scanResults[0].facts.find(
+      (fact) => fact.type === "depGraph",
+    )!.data;
+    expect(depGraph.rootPkg.name).toEqual("docker-image|nginx");
+    expect(depGraph.rootPkg.version).toEqual("1.23.4");
+  });
+
+  it("it overrides name and version when reading an oci archive", async () => {
```

Comment:
Do we think that we need one more case where we have the archive `path` but not the `imageNameAndTag` option?

Reply:
We have this already in regular archives scanning tests (both Docker and OCI) 🙏